applications/mlflow/DEVELOPMENT.md

```diff
@@ -2,42 +2,106 @@
 
 This document provides information about developing, testing, and releasing the MLflow application using the included Taskfile.
 
+## Development
+
+## Prerequisites
+
+- Docker
+- [Task](https://taskfile.dev/installation/) - A task runner / simpler Make alternative
+
+Optional (for direct installation on your machine):
+- go (1.20+)
+- Helm
+- kubectl
+
 ## Development Workflow
 
-The MLflow application includes a Taskfile.yml that provides tasks for developing, testing, and publishing the application.
+We use a [Taskfile](https://taskfile.dev/) to manage development tasks. The recommended approach is to use our containerized development environment, which includes all necessary dependencies.
+
+### Containerized Development Environment
+
+Our Docker-based development environment provides:
+* Consistent environment across team members
+* All required dependencies pre-installed
+* No need to install Go, Python, Helm, or other tools locally
+* Works on any operating system with Docker support
+
+To get started:
+
+```bash
+# Build the development image
+task dev:build-image
+
+# Start a development container
+# This creates and runs the container in the background
+task dev:start
+
+# Enter a shell in an already running development container
+task dev:shell
+```
+
+The tasks above will:
+1. Build the development Docker image if needed
+2. Start a container with the proper mounts and environment
+3. Provide you with a shell inside the container
+4. Allow you to run all task commands directly
+
+### Common Development Tasks
+
+Once you're inside the development container (or on your local machine with all prerequisites installed), you can run these common tasks:
```

Comment:
I'd like to avoid forcing people to operate in the development container, can we just execute the steps in the container instead, or is the shell a convenience thing?

The reason I say this is people are going to be editing and developing files and they will want their normal tools to do that. The container should allow them to run helm commands and go through the workflow but I want to avoid people 'staying' in the container and having to setup their development environment of preference.